connector/src/main/scala/com/basho/riak/spark/util/TSConversionUtil.scala

```diff
@@ -36,27 +36,32 @@ import scala.reflect.ClassTag
 object TimeSeriesToSparkSqlConversion {
   private val STRING_TYPE_REFERENCE = new TypeReference[String] {}
 
-  private def cellValue(sf: StructField, cell: Cell) = sf.dataType match {
-    case BooleanType =>
-      cell.getBoolean
+  private def cellValue(sf: StructField, cellOption: Option[Cell]) = {
+    if (cellOption.isDefined) {
+      val cell = cellOption.get
+      sf.dataType match {
+        case BooleanType =>
+          cell.getBoolean
 
-    case StringType =>
-      cell.getVarcharAsUTF8String
+        case StringType =>
+          cell.getVarcharAsUTF8String
 
-    case LongType =>
-      if (cell.hasTimestamp) cell.getTimestamp else cell.getLong
+        case LongType =>
+          if (cell.hasTimestamp) cell.getTimestamp else cell.getLong
 
-    case IntegerType =>
-      cell.getLong.toInt
+        case IntegerType =>
+          cell.getLong.toInt
 
-    case DoubleType =>
-      cell.getDouble
+        case DoubleType =>
+          cell.getDouble
 
-    case TimestampType =>
-      if (cell.hasLong) new Timestamp(cell.getLong) else new Timestamp(cell.getTimestamp)
+        case TimestampType =>
+          if (cell.hasLong) new Timestamp(cell.getLong) else new Timestamp(cell.getTimestamp)
 
-    case _ =>
-      throw new IllegalStateException(s"Unhandled cell type ${sf.dataType.typeName} for field ${sf.name}")
+        case _ =>
+          throw new IllegalStateException(s"Unhandled cell type ${sf.dataType.typeName} for field ${sf.name}")
+      }
+    } else null
```

Comment:
According to the documentation [Row](https://spark.apache.org/docs/1.4.0/api/java/org/apache/spark/sql/Row.html) is intended to deal with null values:

> It is invalid to use the native primitive interface to retrieve a value that is null, instead a user must check isNullAt before attempting to retrieve a value that might be null.

>  For native primitive access, it is invalid to use the native primitive interface to retrieve a value that is null, instead a user must check isNullAt before attempting to retrieve a value that might be null. An example of native primitive access